demo/fake_server.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Wanda.Executions.FakeServer do
+  @behaviour Wanda.Executions.ServerBehaviour
+
+  import Wanda.Factory
+
+  require Logger
+
+  alias Wanda.Executions
+
+  alias Wanda.Executions.{
+    Execution,
+    Result
+  }
+
+  alias Wanda.Messaging
+
+  @impl true
+  def start_execution(execution_id, group_id, targets, env, config \\ []) do
+    create_fake_execution(execution_id, group_id, targets)
+    Process.sleep(2_000)
+    complete_fake_execution(execution_id, group_id, targets, :passing)
```

Comment:
Can we randomize the result with [https://hexdocs.pm/elixir/1.12/Enum.html#random/1](https://hexdocs.pm/elixir/1.12/Enum.html#random/1) 

You can skip the function parameter and do it directly there

Reply:
@fabriziosestito In our offline conversation, i just was saying, that we can have this execution result coming from the checks results, which now I see is fairly simple. It doesn't need to be random.
You just need to loop through the check results, check the result of them, and simply weight them (or some basic operation to find the worst check result).
Anyway, I'm fine on improving this on future iterations